src/mkdocstrings/handlers/rendering.py

```diff
@@ -146,19 +146,29 @@ def __init__(self, md: Markdown, id_prefix: str):
         self.id_prefix = id_prefix
 
     def run(self, root: Element) -> None:  # noqa: D102 (ignore missing docstring)
-        if not self.id_prefix:
-            return
-        for el in root.iter():
-            id_attr = el.get("id")
-            if id_attr:
-                el.set("id", self.id_prefix + id_attr)
+        if self.id_prefix:
+            self._prefix_ids(root)
+
+    def _prefix_ids(self, root: Element) -> None:
+        index = -1
+        for el in reversed(root):
+            index += 1
 
+            self._prefix_ids(el)
```

Comment:
We descend early. Doing it after handling the element works too. No strong opinion.